app/src/main/java/org/fossasia/susi/ai/adapters/recycleradapters/SearchResultsAdapter.java

```diff
@@ -44,54 +44,21 @@ public SearchResultHolder onCreateViewHolder(ViewGroup parent, int viewType) {
 
     @Override
     public void onBindViewHolder(final SearchResultHolder holder, int position) {
-        Datum datum = datumList.get(position);
+        final Datum datum = datumList.get(position);
+        holder.previewImageView.setVisibility(View.GONE);
         if (datum != null) {
             holder.titleTextView.setText(Html.fromHtml(datum.getTitle()));
             holder.descriptionTextView.setText(Html.fromHtml(datum.getDescription()));
-            if (!TextUtils.isEmpty(datum.getLink())) {
-                LinkPreviewCallback linkPreviewCallback = new LinkPreviewCallback() {
-                    @Override
-                    public void onPre() {
-                        holder.previewImageView.setVisibility(View.GONE);
-                        holder.descriptionTextView.setVisibility(View.GONE);
-                        holder.titleTextView.setVisibility(View.GONE);
-                        holder.previewLayout.setVisibility(View.GONE);
+            holder.previewLayout.setOnClickListener(new View.OnClickListener() {
+                @Override
+                public void onClick(View view) {
+                    Uri webpage = Uri.parse(datum.getLink());
+                    Intent intent = new Intent(Intent.ACTION_VIEW, webpage);
+                    if (intent.resolveActivity(context.getPackageManager()) != null) {
+                        context.startActivity(intent);
                     }
-
-                    @Override
-                    public void onPos(final SourceContent sourceContent, boolean b) {
-                        holder.previewLayout.setVisibility(View.VISIBLE);
-                        holder.previewImageView.setVisibility(View.VISIBLE);
-                        holder.descriptionTextView.setVisibility(View.VISIBLE);
-                        holder.titleTextView.setVisibility(View.VISIBLE);
-                        holder.titleTextView.setText(sourceContent.getTitle());
```

Comment:
@chiragw15 @betterclever here text is added in textview from websearch result by linkpreviewcallback. I initially tried app without removing this code but it gives result like this.

![screenshot_2017-06-13-13-28-09](https://user-images.githubusercontent.com/14003244/27074654-66b955f2-5045-11e7-91dc-850614eebc48.png)
can you also check this? and why we need preview. In rss we have to use result from server. There is a link in server we use that link to show result in browser.